### PR TITLE
Update Lesson 4 homework answer

### PR DESCRIPTION
**Pull request by YuelinYang**
- [ ] This PR contains only information intended for public view

To view this PR's branch in binder:
```
https://mybinder.org/v2/gh/viettrung9012/python-codeacademy-sg/<branch>?urlpath=lab
```

### CODE REVIEW

File: Homework/Solutions/Lesson4.ipynb [hunk 1]
```diff
@@ -0,0 +1,257 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "## Some Takehome Exercises (Opitonal) ---- Solutions\n",
```
Comment (kennytean) [inline Homework/Solutions/Lesson4.ipynb]: Optional

File: Homework/Solutions/Lesson4.ipynb [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+   "metadata": {},
+   "source": [
+    "<span>2. </span>\n",
+    "a) Write a function to calculate the accumulated number of steps from day 0 to day x for each participant.\n",
+    "Example: Print the number of steps for each participant from day 1 to day 10.\n",
+    "Sample output:\n",
+    "```\n",
+    "Participants 1's steps until day 7: 134267\n",
+    "Participants 2's steps until day 7: 146012\n",
+    "Participants 3's steps until day 7: 139920\n",
```
Comment (FabienLauf) [inline Homework/Solutions/Lesson4.ipynb]: "Participant" without "s"

File: Homework/Solutions/Lesson4.ipynb [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    "...\n",
+    "```\n",
+    "b) Integrate with the \"input\" concept from Lesson 2, input the day that you want to calculate until, i.e. the cutoff day, and call the function you wrote in 2a.\n",
+    "Sample output:\n",
+    "```\n",
+    "Enter the cut-off day: 7\n",
+    "Participants 1's steps until day 7 : 134267\n",
+    "Participants 2's steps until day 7 : 146012\n",
+    "Participants 3's steps until day 7 : 139920\n",
```
Comment (FabienLauf) [inline Homework/Solutions/Lesson4.ipynb]: "Participant" without "s"

File: Homework/Solutions/Lesson4.ipynb [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    "...\n",
+    "```\n",
+    "c) Reusing the function written in 2a, print all situations where the participants' accumulated number of steps exceeds 400,000. (Hint: calculate the number of accumulated steps every day.)\n",
+    "Sample output:\n",
+    "```\n",
+    "Participants 38 on day 24 walked 406222\n",
+    "Participants 3 on day 25 walked 408190\n",
```
Comment (FabienLauf) [inline Homework/Solutions/Lesson4.ipynb]: "Participant" without "s"

File: Homework/Solutions/Lesson4.ipynb [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+    "with open('../../Biggest Loser 2018.csv') as csvfile: # IMPT: if you are running the code from the original location of Lesson 4, change '../../Biggest Loser 2018.csv' to 'Biggest Loser 2018.csv'\n",
+    "    readCSV = csv.reader(csvfile)\n",
+    "    stepsData = list(readCSV) # The list method convert the reader item - readCSV to a list of lists (array of arrays)\n",
+    "    total_days = len(stepsData[0]) - 4 # total number of columns minus the four coloums that do not contain steps information\n",
+    "    for i in range(1, total_days + 1): # iterate from day 1 to the end of the competition\n",
+    "        for index, row in enumerate(stepsData): \n",
+    "            if index > 0:\n",
+    "                total_steps = total_steps_by_day(row, i)\n",
+    "                if total_steps > 400000:\n",
+    "                    print(\"Participants \" + str(index) + \" on day \" + str(i) + \" walked \" + str(total_steps))"
```
Comment (FabienLauf) [inline Homework/Solutions/Lesson4.ipynb]: "Participant" without "s"

**Review by FabienLauf — CHANGES_REQUESTED** [review]

**Review by FabienLauf — APPROVED** [review]

**Pull request merged**